### PR TITLE
do not drop group control message from blocked user

### PR DESCRIPTION
**Pull request by Bilb**
We block a message from a user in all cases except if:
* this message is a group message
* the group already exist in our db
* the group is not blocked
* the content of the message is control only

### CODE REVIEW

File: ts/receiver/contentMessage.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { getEnvelopeId } from './common';
 
 import { removeFromCache, updateCache } from './cache';
 import { SignalService } from '../protobuf';
-import { toNumber } from 'lodash';
+import { identity as lodashIdentiy, isEmpty, pickBy, toNumber } from 'lodash';
```
Comment (vincentbavitz) [inline ts/receiver/contentMessage.ts:7]: Cleaner to use `Lodash.identity`, etc?

File: ts/receiver/contentMessage.ts [hunk 2]
```diff
@@ -344,6 +383,17 @@ export async function innerHandleContentMessage(
 
   const content = SignalService.Content.decode(new Uint8Array(plaintext));
 
+  const blocked = await isBlocked(envelope.source);
+  if (blocked) {
+    // we want to allow a blocked user message if that's a control message for a known group and the group is not blocked
```
Comment (vincentbavitz) [inline ts/receiver/contentMessage.ts:388]: `// We want... `

File: ts/receiver/contentMessage.ts (hunk 1 above)
Comment (Bilb) [reply ts/receiver/contentMessage.ts]: amended

File: ts/receiver/contentMessage.ts (hunk 2 above)
Comment (Bilb) [reply ts/receiver/contentMessage.ts]: fixed

**Pull request merged**